### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Spelling fixes

## Description
Semi-automated spelling review of codebase (includes comments, code, public apis, documentation, etc)

## Motivation and Context
I have a toolchain which I've been evolving for a while, and I like to test it on projects, especially as a way to give a little in appreciation of projects that I may be using/have used/plan to use.

## How Has This Been Tested?
- [x] This has not been tested.

## Types of change
- [x] Breaking change
- - [x] Public APIs - I'm pretty sure I've renamed some public APIs. I'm happy to split such items out (either including backwards compat or dropping them entirely), but I'd rather have a conversation about which those things are before I eagerly split.
- - [x] Command line flags (probably)
- - [x] Console output
- - [x] Debug ouput
- - [x] CSS identifiers/rules
- [x] Comments
- [x] Local variables
- [x] Local functions
- [x] Class member and static variables
- [x] Doxygen content
- [x] English translations (but not other languages)
- [x] Addons (if they are maintained elsewhere, let me know)
- [x] Spelling only (i.e. not grammar, not even to make things agree)
- [x] No effort has been made to adjust whitespace based on the length of replacement words
- [x] External modules have been excluded to the extent that I could identify them. If I've missed things, I'll gladly toss them out, please identify them.

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
Well sorry but looks like totally unacceptable to merge it like this

File: addons/resource.language.en_gb/resources/langinfo.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
 <!-- CP1255 - Hebrew (Windows)                          -->
 <!-- CP1256 - Arabic (Windows)                          -->
 <!-- CP1257 - Baltic (Windows)                          -->
-<!-- CP1258 - Vietnamesse (Windows)                     -->
+<!-- CP1258 - Vietnamese (Windows)                     -->
```
Comment (AlwinEsch) [inline addons/resource.language.en_gb/resources/langinfo.xml]: Can you put a new spacer to leave `-->`.

But until now your changes looks very good.

**Comment by jsoref** [comment]
I'm happy to split things out. Just describe how you want them split.

I can certainly start w/ comments only. The reason the commits are the way they are is that it's easiest for me to manage rebases this way -- there was about a month's drift between when I started w/ your repo and when I was able to finish.

The best starting point would be for someone to do a git diff --stat and categorize the files.
Before I do more work, I need a rough outline of:
 1. These files are from external sources / shouldn't be changed
 1. Please split these files into distinct commits
 1. Please provide PRs according to {list of criteria}

Sample criteria include:
 * Comments
 * Debug strings
 * Console Output
 * Commandline input
 * Local variables
 * Local functions
 * Class variables
 * Classnames
 * Public functions
 * CSS
 * Other markup
 * Doxygen / other documentation

Again, I'm willing to do the split. But I'm not going to guess what split you want. Nor do I realistically have the ability to determine which files are "internal" to your project or "external" to your project -- beyond a first attempt that I've already made.

I can talk to Travis and AppVeyor to get them to green while I wait for input.

File: addons/resource.language.en_gb/resources/langinfo.xml (hunk 1 above)
Comment (jsoref) [inline addons/resource.language.en_gb/resources/langinfo.xml]: Yeah, I can do that

**Comment by jsoref** [comment]
Sorry about the build fail. I had fixed the "top side" but neglected to drop the bottom side for one file. Travis should be happier now...

File: addons/resource.language.en_gb/resources/langinfo.xml (hunk 1 above)
Comment (jsoref) [inline addons/resource.language.en_gb/resources/langinfo.xml]: I've done this locally, but I don't plan to push until I receive a bunch of additional comments.

**Comment by AlwinEsch** [comment]
Your changes are very good, but @MartijnKaijser is right. To bring in with the amount of commits make it not possible to identify the related places. Is not critical if on a comment of code or text documents, but on changes from code itself, must we see exactly where. Otherwise we can not make sure if some external parts who use the same spelling are still compatible.

**Comment by jsoref** [comment]
I just need some hint about how to split it... I can certainly do a first round that's just comments...

**Comment by jsoref** [comment]
I've split out most of the comments as #11348

**Comment by DaveTBlake** [comment]
Since the changes are being split into separate PRs (a good thing) shouldn't this PR itself be closed?

**Comment by jsoref** [comment]
It can be, I think it's vaguely helpful to have a place for people to see a roadmap. But it's your tracker, so if you want to close it, I won't object.

**Comment by jsoref** [comment]
I think for now this is still a rough guide, and it looks like it's probably best to merge the other split out commits. We're in the home stretch.

**Comment by MartijnKaijser** [comment]
Is everything merged by now?

**Comment by jsoref** [comment]
This PR has two commits left, that's all. I can split them into a PR or two if you want...

**Comment by MartijnKaijser** [comment]
Not sure about eventclients cause that's actually a third party library iirc. 
Jenkins build this please

**Comment by garbear** [comment]
@AlwinEsch how are binary add-ons looking? is this version stuff still used in the agile branch?

**Comment by garbear** [comment]
This breaks ABI for some game add-ons. Not against this, but it needs its own PR so we can track down all the places this is used.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_dll.h [hunk 2]
```diff
@@ -46,7 +46,7 @@ const char* GetGameAPIVersion(void);
  *
  * \return Must be GAME_MIN_API_VERSION_STRING
  */
-const char* GetMininumGameAPIVersion(void);
+const char* GetMinimumGameAPIVersion(void);
```
Comment (garbear) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_dll.h]: This breaks ABI for some game add-ons. Not against this, but it needs its own PR so we can track down all the places this is used.

Comment (MartijnKaijser) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_dll.h]: @jsoref if you can split this out i guess we're almost ready :)

Comment (Razzeee) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_dll.h]: @jsoref can you do a seperate PR for the commit you split out here?

Comment (jsoref) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_dll.h]: Sure

Comment (Razzeee) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_dll.h]: Kind reminder :)

Comment (garbear) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_dll.h]: broken out to: https://github.com/xbmc/xbmc/pull/11993

**Comment by Razzeee** [comment]
@ksooo  for PVR, probably?

Not sure who to ping about GUI?

**Comment by AlwinEsch** [comment]
The spelling correction is no more needed. Related functions are with https://github.com/xbmc/xbmc/pull/12109 removed.

**Pull request closed without merging**